### DISCUSSION TITLE
Custom DxcIncludeHandler

### DISCUSSION THREAD

**Discussion by tomdrin88** [amerkoleci/Vortice.Windows#153] [Q&A] [answered]
Hello, 
I would like to implement my own include handler - basically prepending the path with custom folder. I have written simple implementation - see the code below.
It seems to produce correct blob if I call it outside the compiler scope. But unfortunatelly once I use it in DxcCompiler.Compile I get file not found error - it seems like it is ignored completelly.

Is there some tricky part I'm missing?
Is somewhere example of custom IncludeHandler?

        private static readonly IDxcUtils UTILS = Dxc.CreateDxcUtils();

        private class IncludeHandler : SharpGen.Runtime.ComObject, IDxcIncludeHandler
        {
            public Result LoadSource(string filenameRef, out IDxcBlob includeSourceOut)
            {
                includeSourceOut = null;
                
                var path = Path.Combine(ShadersBasePath, filenameRef);
                
                if (!File.Exists(path))
                    return Result.Fail;
                
                includeSourceOut = UTILS.LoadFile(path, null);
                return Result.Ok;
            }
        }

Thanks a lot for the answer in advance
Tomas Drinovsky

**Comment by amerkoleci** [comment, marked as answer]
Hi, nothing tricky.

Just grab code from here:
https://github.com/amerkoleci/Vortice.Windows/blob/main/src/samples/HelloDirect3D12/ShaderIncludeHandler.cs

**Reply by tomdrin88** [reply]
Thanks a lot :)
